src/roswire/common/package.py

```diff
@@ -204,7 +159,7 @@ def _from_paths(cls,
                     raise
             else:
                 packages.append(package)
-        return PackageDatabase(packages, paths)
+        return cls(packages, paths)
```

Comment:
This assumes knowledge of the constructor for `PackageDatabase` subclasses. We should probably add and use a abstract class method `for_packages_and_paths(packages, paths)` here instead.